File: converted/src/zombie/text/templating/IReplaceProvider.cpp
#include "zombie/text/templating/IReplaceProvider.h"

namespace zombie {
namespace text {
namespace templating {

bool IReplaceProvider::hasReplacer(const std::string &key) {
  // TODO: Implement hasReplacer
  return false;
}

IReplace IReplaceProvider::getReplacer(const std::string &key) {
  // TODO: Implement getReplacer
  return nullptr;
}

} // namespace templating
} // namespace text
} // namespace zombie

File: source/zombie/text/templating/IReplaceProvider.java
// Decompiled on Sat Jan 17 08:24:00 EST 2026 with Zomboid Decompiler v0.2.3 using Vineflower.
package zombie.text.templating;

/**
 * TurboTuTone.  Interface that can provide IReplace for ITemplateBuilder.  Any keys present in a IReplaceProvider take priority when replacing keys in a template.   NOTE: When checking the key String, this should be equaling a lowercase version or equalsIgnoreCase
 */
public interface IReplaceProvider {
    boolean hasReplacer(String key);

    IReplace getReplacer(String key);
}
